File: go/problems/day12_test.go
package problems

import "testing"

var day12 Solvable[int] = Day12{}

func TestDay12_TestSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day12_1")
	result := day12.Solve(input)

	var expected int = 21
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

func TestDay12_FullSet_ShouldYield_Result(t *testing.T) {
	input := readProblemInput("day12_2")
	result := day12.Solve(input)

	var expected int = 7017
	if result != expected {
		t.Errorf("expected %d but got %d", expected, result)
	}
}

File: ts/problems/day12.test.ts
import test, { describe } from "node:test";
import assert from "node:assert";

import solve from "./day12.js";
import { readProblemInput } from "../inputReader.js";

describe("Day 12", () => {
  test("test set should yield expected result", () => {
    const input = readProblemInput("day12_1");
    const result = solve(input);
    assert.equal(result, 21);
  });

  test("full set should yield expected result", () => {
    const input = readProblemInput("day12_2");
    const result = solve(input);
    assert.equal(result, 7017);
  });
});
